packages/@ngtools/webpack/src/loader.ts

```diff
@@ -248,14 +225,72 @@ function _replaceBootstrap(plugin: AotPlugin, refactor: TypeScriptFileRefactor)
     // Add the appropriate import
     refactor.insertImport(platform.name, platform.importLocation);
   });
+}
+
+
+function _replaceBootstrap(refactor: TypeScriptFileRefactor) {
+  // If bootstrapModule can't be found, bail out early.
+  if (!refactor.sourceMatch(/\bbootstrapModule\b/)) {
+    return;
+  }
+
+  const bootstraps = refactor.findAstNodes(refactor.sourceFile, ts.SyntaxKind.CallExpression, true)
+    .filter((call: ts.CallExpression) =>
+      call.expression.kind == ts.SyntaxKind.PropertyAccessExpression)
+    .map((call: ts.CallExpression) => call.expression as ts.PropertyAccessExpression)
+    .filter(access => {
+      return access.name.kind == ts.SyntaxKind.Identifier
+        && access.name.text == 'bootstrapModule';
+    });
+
+  if (bootstraps.length == 0) {
+    // Didn't find any dynamic bootstrapping going on.
+    return;
+  }
+
+  _replacePlatform(refactor, bootstraps);
 
   bootstraps
     .forEach((bs: ts.PropertyAccessExpression) => {
       // This changes the call.
       refactor.replaceNode(bs.name, 'bootstrapModuleFactory');
     });
+}
+
+
+function _replaceEntryModule(plugin: AotPlugin, refactor: TypeScriptFileRefactor) {
+  const modules = refactor.findAstNodes(refactor.sourceFile, ts.SyntaxKind.Identifier, true)
```

Comment:
Seems like that would catch any identifier that's the same as the class name as you don't look at the source of the identifier, and the original source took into account code like this:

```typescript
import {AppModule} from './app/app.module';
// ... other imports

AppModule.someStaticProperty = 1;  // This would now fail with this code, but worked with the original.

// bootstrapping occurs.
```

We need to make sure this use case still work. I'm pretty sure this is a more common use case than we think. Is there a reason you replace ALL mentions of the identifier, regardless of whether it's in the bootstrapping call or not?